hera_cal/io.py

```diff
@@ -7,26 +7,537 @@
+            else:
+                raise TypeError('If input_data is a list, it must be a list of strings.')
+        else:
+            raise ValueError('input_data must be a string or a list of strings.')
+        for f in self.filepaths:
+            if not os.path.exists(f):
+                raise IOError('Cannot find file ' + f)
+
+        # load metadata from file
+        self.filetype = filetype
+        if self.filetype is 'uvh5':
+            # read all UVData metadata from first file
+            temp_paths = copy.deepcopy(self.filepaths)
+            self.filepaths = self.filepaths[0]
+            self.read(read_data=False)
+            self.filepaths = temp_paths
+
+            if len(self.filepaths) > 1:  # save HERAData_metas in dicts
+                for meta in self.HERAData_metas:
+                    setattr(self, meta, {})
+                for path in self.filepaths:
+                    hc = HERAData(path, filetype='uvh5')
+                    meta_dict = self.get_metadata_dict()
+                    for meta in self.HERAData_metas:
+                        getattr(self, meta)[path] = meta_dict[meta]
+            else:  # save HERAData_metas as attributes
+                self._writers = {}
+                for key, value in self.get_metadata_dict().items():
+                    setattr(self, key, value)
+
+        elif self.filetype in ['miriad', 'uvfits']:
+            for meta in self.HERAData_metas:
+                setattr(self, meta, None)  # no pre-loading of metadata
+        else:
+            raise NotImplementedError('Filetype ' + self.filetype + ' has not been implemented.')
+
+    def reset(self):
+        '''Resets all standard UVData attributes, potentially freeing memory.'''
+        super(HERAData, self).__init__()
+
+    def get_metadata_dict(self):
+        ''' Produces a dictionary of the most useful metadata. Used as object
+        attributes and as metadata to store in DataContainers.
+
+        Returns:
+            metadata_dict: dictionary of all items in self.HERAData_metas
+        '''
+        antpos, ants = self.get_ENU_antpos()
+        ants = sorted(ants)
+        antpos = dict(zip(ants, antpos))
+
+        freqs = np.unique(self.freq_array)
+        times = np.unique(self.time_array)
+        lst_indices = np.unique(self.lst_array.ravel(), return_index=True)[1]
+        lsts = self.lst_array.ravel()[np.sort(lst_indices)]
+        pols = [polnum2str(polnum) for polnum in self.polarization_array]
+        antpairs = self.get_antpairs()
+        bls = [antpair + (pol,) for antpair in antpairs for pol in pols]
+
+        times_by_bl = {antpair: np.array(self.time_array[self._blt_slices[antpair]])
+                       for antpair in antpairs}
+        lsts_by_bl = {antpair: np.array(self.lst_array[self._blt_slices[antpair]])
+                      for antpair in antpairs}
+
+        locs = locals()
+        return {meta: eval(meta, {}, locs) for meta in self.HERAData_metas}
```

Comment:
I hear its not good practice to use the built-in `eval` function, as its not safe. Perhaps try
```
locs = locals()
return odict([(meta, locs[meta]) for meta in self.HERAData_metas])
```
just a suggestion though...